connection.go

```diff
@@ -10,13 +10,25 @@ type GormConnection struct {
 	db *gorm.DB
 }
 
-func NewPostgresConnection(params *DBConnectionParams) (*GormConnection, error) {
+func NewPostgresConnection(params *DBConnectionParams, config *DBConnectionConfig) (*GormConnection, error) {
 	dsn, err := params.GetDSN()
 	if err != nil {
 		return nil, fmt.Errorf("failed to retrieve dsn")
 	}
 
-	conn, err := gorm.Open(postgres.Open(dsn), &gorm.Config{})
+	conn, err := gorm.Open(postgres.Open(dsn), &gorm.Config{
```

Comment:
Are we sure we aren't overriding all default config values with this? (As in, changing any missing params to their Go type's default instead of the GORM default value)
Ex. if any of them are "true" by default and the caller does not explicitely pass a value for that param, will they be set to "false"?

[GORM seems to check for nils](https://github.com/go-gorm/gorm/blob/912360097a2f54bb0f0ee4b02f9b39c591071837/gorm.go#L131), but can we make sure?

Reply:
can we check this before merging?

Reply:
Ooops, I hit merge without checking this. I reverted the merge process and now I will check it first.